### PR TITLE
Added type definitions for bodyBuilder

### PR DESCRIPTION
**Pull request by islimane**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,8 @@
     "types-publisher": "Microsoft/types-publisher#production"
   },
   "dependencies": {
-    "jslint": "^0.10.3"
+    "bodybuilder": "^2.2.0",
```
Comment (weswigham) [inline package.json]: Revert edits to the root `package.json` - if your package depends on type from another not-DT package, then your package's folder should have a `package.json` listing those type dependencies. The `test` file isn't for runtime tests - it's for typechecker tests - making sure objects typecheck correctly.

File: bodybuilder/bodybuilder.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Type definitions for bodybuider v2.2.0
+// Project: https://github.com/danpaz/bodybuilder
+// Definitions by: Ismael Slimane Zubillaga <https://github.com/islimane>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module "bodybuilder" {
```
Comment (weswigham) [inline bodybuilder/bodybuilder.d.ts]: Rename this file to `index.d.ts`, and ideally rewrite the type definition to
```ts
declare function bodybuilder(): any;
export = bodybuilder; 
```
But I will heartily recommend more useful type definitions!

File: bodybuilder/bodybuilder-tests.ts [hunk 3]
```diff
@@ -0,0 +1,708 @@
+import test from 'tape';
```
Comment (weswigham) [inline bodybuilder/bodybuilder-tests.ts]: Rewrite this file to test type compatibility, not runtime behavior.

**Review by weswigham — CHANGES_REQUESTED** [review]
In addition to the inline comments, please add a `tslint.json` with the contents `{ "extends": "dtslint/dt.json" }` and run the lint.

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has failing CI tests and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**